internal/services/automation/automation_runbook_resource.go

```diff
@@ -204,13 +204,22 @@ func resourceAutomationRunbookCreateUpdate(d *pluginsdk.ResourceData, meta inter
 		reader := io.NopCloser(bytes.NewBufferString(content))
 		draftClient := meta.(*clients.Client).Automation.RunbookDraftClient
 
-		if _, err := draftClient.ReplaceContent(ctx, id.ResourceGroup, id.AutomationAccountName, id.Name, reader); err != nil {
+		_, err := draftClient.ReplaceContent(ctx, id.ResourceGroup, id.AutomationAccountName, id.Name, reader)
+		if err != nil {
 			return fmt.Errorf("setting the draft for %s: %+v", id, err)
 		}
+		// Uncomment below once https://github.com/Azure/azure-sdk-for-go/issues/17196 is resolved.
+		// if err := f1.WaitForCompletionRef(ctx, draftClient.Client); err != nil {
+		// 	return fmt.Errorf("waiting for set the draft for %s: %+v", id, err)
+		// }
```

Comment:
should we be polling on something else? e.g. until the `provisioningState` is `Succeeded`?

Reply:
There is no `provisioningState` in the polling response (actually there is no response body). Also, a `GET` call against the `.../draft/content` simply returns what you have sent in the `PUT`. So this seems like a sync API but incorrectly marked as LRO.

Reply:
In this case the `provisioningState` would be against the parent Automation Runbook rather than this API - but that should be fine 👍